src/bp/ui-admin/src/telemetry/offlineServerFallback.ts

```diff
@@ -0,0 +1,55 @@
+import axios from 'axios'
+import _ from 'lodash'
+import ms from 'ms'
+
+import api from '../api'
+
+import { axiosConfig } from './client'
+
+const serverUrlPayloads = '/admin/telemetry-payloads'
+const serverUrlFeedback = '/admin/telemetry-feedback'
+
+const sendServerPackage = async () => {
+  let continueLoop = true
+  do {
+    try {
+      const {
+        data: { events }
+      } = await api.getSecured().get(serverUrlPayloads)
+
+      if (_.isEmpty(events)) {
+        break
+      }
+
+      const status = await postStorageServer(events)
+      continueLoop = status === 'ok'
+
+      await api.getSecured().post(serverUrlFeedback, { events: events.map(e => e.uuid), status })
+    } catch (err) {
+      console.error('Could not access the botpress server', err)
+
+      continueLoop = false
+    }
+  } while (continueLoop)
+}
+
+const postStorageServer = async events => {
+  let status
+  try {
+    await axios.post(
+      '/',
+      events.map(e => ({ ...e, source: 'client' })),
+      axiosConfig
+    )
```

Comment:
Pro tip: to have your request beautifully on a single line, move the events.map method to a const  before your axios call.

Your status variable is redundant, just return 'ok' or return 'fail' in either case...

Reply:
Fixed in : 
1a22377